bears/general/InvalidLinkBear.py

```diff
@@ -49,8 +49,33 @@ def parse_pip_vcs_url(link):
     def find_links_in_file(file, timeout, link_ignore_regex):
         link_ignore_regex = re.compile(link_ignore_regex)
         regex = re.compile(
-            r'((git\+|bzr\+|svn\+|hg\+|)https?://[^.:%\s_/?#[\]@\\]+\.(?:[^\s('
-            r')%\'"`<>|\\]+|\([^\s()%\'"`<>|\\]*\))*)(?<!\.)(?<!,)')
+            r"""
+            ((git\+|bzr\+|svn\+|hg\+|)  # For VCS URLs
+            https?://                   # http:// or https:// as only these
+                                        # are supported by the ``requests``
+                                        # library
+            [^.:%\s_/?#[\]@\\]+         # Initial part of domain
+            \.                          # A required dot `.`
+            (
+                (?:[^\s()%\'"`<>|\\]+)  # Path name
+                                        # This part does not allow
+                                        # any parenthesis: balanced or
+                                        # unbalanced.
+            |                           # OR
+                \([^\s()%\'"`<>|\\]*\)  # Path name contained within ()
```

Comment:
> There can be, but we cant detect it with this regex. I.e. the regex is sub-optimal, but improving it is hard without introducing false positives.

Actually WRT this I disagree and think the behaviour is already pretty optimal: unbalanced parentheses are likely part of markup around the URL. The whole truth is that the whole URL detection is a heuristic because otherwise we'd have to understand the markup language which we ATM don't. So it does things like this and so far this has yielded a very very low false positive rate.

(I wrote the regex mostly I think FWIW.)